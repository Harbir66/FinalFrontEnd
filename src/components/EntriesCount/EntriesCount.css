.entries-count{
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 19vw;
    margin: 5vh 3vw 0;
    color: #4F5054;
}

.small{
    font-size: large;
}

.small>img{
    height: 3vh;
}

.large{
    margin: 0;
    font-size: 2.5vh;
    color: #8A8C93;
}