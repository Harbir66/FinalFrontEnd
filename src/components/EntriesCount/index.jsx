import React from 'react';
import './EntriesCount.css';
import PropTypes from 'prop-types';
import searchIcon from '../../assets/icon-search-dark@3x.png';

function EntriesCount({ count, type, showSearch }) {
  const size = showSearch ? 'small' : 'large';
  return (
    <div className={`entries-count ${size}`}>
      <span>{`${count} ${type}`}</span>
      {showSearch && (
        <img src={searchIcon} alt="searchIcon" className="search-icon" />
      )}
    </div>
  );
}

export default EntriesCount;

EntriesCount.propTypes = {
  count: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
  showSearch: PropTypes.bool.isRequired,
};
