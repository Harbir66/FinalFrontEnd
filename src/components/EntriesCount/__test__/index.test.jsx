import React from 'react';
import { render } from '@testing-library/react';
import EntriesCount from '..';

describe('EntriesCount', () => {
  it('should render correctly with search icon', () => {
    const { asFragment } = render(
      <EntriesCount count={10} type="mockType" showSearch />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render correctly without search icon', () => {
    const { asFragment } = render(<EntriesCount count={10} type="mockType" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
