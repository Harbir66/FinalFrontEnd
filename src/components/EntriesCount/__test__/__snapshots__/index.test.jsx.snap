// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntriesCount should render correctly with search icon 1`] = `
<DocumentFragment>
  <div
    class="entries-count small"
  >
    <span>
      10 mockType
    </span>
    <img
      alt="searchIcon"
      class="search-icon"
      src="icon-search-dark@3x.png"
    />
  </div>
</DocumentFragment>
`;

exports[`EntriesCount should render correctly without search icon 1`] = `
<DocumentFragment>
  <div
    class="entries-count large"
  >
    <span>
      10 mockType
    </span>
  </div>
</DocumentFragment>
`;
